chroot/run.go

```diff
@@ -84,9 +84,17 @@ type runUsingChrootExecSubprocOptions struct {
 // RunUsingChroot runs a chrooted process, using some of the settings from the
 // passed-in spec, and using the specified bundlePath to hold temporary files,
 // directories, and mountpoints.
-func RunUsingChroot(spec *specs.Spec, bundlePath string, stdin io.Reader, stdout, stderr io.Writer) (err error) {
+func RunUsingChroot(spec *specs.Spec, bundlePath, homeDir string, stdin io.Reader, stdout, stderr io.Writer) (err error) {
 	var confwg sync.WaitGroup
-
+	var homeFound bool
+	for _, env := range spec.Process.Env {
+		if strings.HasPrefix(env, "HOME=") {
+			homeFound = true
+		}
+	}
+	if !homeFound {
+		spec.Process.Env = append(spec.Process.Env, fmt.Sprintf("HOME=%s", homeDir))
```

Comment:
Move this to the line where you find HOME= and then remove homeFound variable.

Reply:
I don't think I can.  I only want to add the variable if it's not found.  If I move it to 'homeFound = true' (line 92), I'd be adding the envvar only when it existed.  We have to go through the entire slice to make sure it's not there.

Reply:
Ok I read it wrong. I thought you were adding when it was found...